elastic/kibana
# [Synthetics] Fix MTLS settings for synthetics service in Kibana

## Summary

This PR fixes the MTLS settings for the service in Kibana which broke due to recent changes in how we used `devUrl` and other settings to decide how to send requests.

Now, we'll always `rejectUnauthorized` certs unless users are sending requests to `localhost` and in `dev` mode, it doesn't matter whether they serve their own manifests or use `devUrl`.


### Risk Matrix

* Please review the certificate settings carefully to see if there's anything I missed here which would allow for unauthorised certs out of dev and `localhost`. 


### How to test this PR

1. Checkout the latest version of the service (make sure it contains the recent changes to DX due to Makefile updates)
2. Run the service as per the instructions on "how to run it locally" that are present in the docs
3. Setup Kibana to use the service using `devUrl` and `basicAuth`. Ensure you can create and run monitors (try to use run-once as it's easier to test).
4. Setup Kibana to use the service using `devUrl` and certificates by copying the `crt` and `key` in the service's folder to the Kibana `config` folder and using it in the Kibana config. For example:
    ```yaml
    # Make sure you do NOT have username and password setup for basic auth
    xpack.uptime.service.devUrl: https://localhost:10001
    xpack.uptime.service.tls.certificate: config/123-client.pem
    xpack.uptime.service.tls.key: config/123-client.key
    ```
    Ensure you can create and run monitors (try to use run-once as it's easier to test).
5. Setup Kibana to use a local manifest by creating a manifest file like the one below and serving it on port 8081 using `npx http-server --port 8081`.
    ```json
    {
      "throttling": {
        "download": 20,
        "upload": 10
      },
      "locations": {
        "local": {
          "url": "https://localhost:10001",
          "geo": {
            "name": "My Test [local]",
            "location": {"lat": 41.25, "lon": -95.86}
          },
          "status": "ga"
        }
      }
    }
    ```

    Then update Kibana configs to use the served manifest:

    ```yaml
    # Comment the devUrl!
    # xpack.uptime.service.devUrl: https://localhost:10001
    
    xpack.uptime.service.manifestUrl: http://localhost:8081/manifest.json
    
    # Make sure you do NOT have username and password setup for basic auth
    xpack.uptime.service.tls.certificate: config/123-client.pem
    xpack.uptime.service.tls.key: config/123-client.key
    ```
    Ensure you can create and run monitors (try to use run-once as it's easier to test).


### For maintainers
